components/camel-kafka/src/test/java/org/apache/camel/component/kafka/integration/KafkaProducerFullIT.java

```diff
@@ -155,10 +175,60 @@ public void configure() {
                 from("direct:propagatedHeaders").to(toPropagatedHeaders).to(mockEndpoint);
 
                 from("direct:noRecordSpecificHeaders").to(toNoRecordSpecificHeaders).to(mockEndpoint);
+
+                from("direct:startTransaction").transacted("kafkaTransaction")
+                        .to(toTransaction).process(new Processor() {
+                            @Override
+                            public void process(Exchange exchange) throws Exception {
+                                String body = exchange.getIn().getBody(String.class);
+                                if (body.contains("fail")) {
+                                    throw new RuntimeException("fail process message " + body);
+                                }
+                            }
+                        }).to(mockEndpoint);
             }
         };
     }
 
+    @Test
+    public void producedTransactionMassageIsReceivedByKafka() throws InterruptedException {
+        int messageInTopic = 10;
+        int messageInOtherTopic = 5;
+
+        CountDownLatch messagesLatch = new CountDownLatch(messageInTopic + messageInOtherTopic);
+
+        sendMessagesInRoute(messageInTopic, testTransaction, "IT test transaction message", KafkaConstants.PARTITION_KEY, "0");
+        try {
+            sendMessagesInRoute(messageInTopic, testTransaction, "IT test transaction fail message",
+                    KafkaConstants.PARTITION_KEY,
+                    "0");
+            fail("Should fail in transaction");
+        } catch (Exception e) {
+        }
```

Comment:
Please use `assertThrows` or `assertDoesNotThrow` depending on what you expect the code to do.

Reply:
good catch!

Reply:
Done